tests/test_metadata_handling.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+import yaml
+from shaker import helpers
+from nose.tools import raises
+
+class TestMetadataHandling(unittest.TestCase):
+
+    # Sample metadata with duplicates
+    _sample_metadata_duplicates = {
+        "dependencies": [
+            "git@github.com:test_organisation/test1-formula.git==v1.0.1",
+            "git@github.com:test_organisation/test1-formula.git==v1.0.1",
```

Comment:
I have a feeling it was a duplicate with the same version and something nasty happens in the recursive logic just by processing it twice.